WordPress/Classes/Services/MediaService.m

```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
                 failure:failureBlock];
 }
 
+- (NSError *)translateMediaUploadError:(NSError *)error {
+    NSError *newError = error;
+    if (error.domain == WordPressComApiErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case WordPressComApiErrorUploadFailed:
+                errorMessage = NSLocalizedString(@"The app couldn't upload this media.", @"Message to show to user when media upload failed by unknow server error");
+                errorCode = WordPressComApiErrorUploadFailed;
+                break;
+            case WordPressComApiErrorUploadFailedInvalidFileType:
+                errorMessage = NSLocalizedString(@"Your site does not support this media file format.", @"Message to show to user when media upload failed because server doesn't support media type");
+                errorCode = WordPressComApiErrorUploadFailedInvalidFileType;
+                break;
+            case WordPressComApiErrorUploadFailedNotEnoughDiskQuota:
+                errorMessage = NSLocalizedString(@"Your site is out of space for media uploads.", @"Message to show to user when media upload failed because user doesn't have enough space on quota/disk");
+                errorCode = WordPressComApiErrorUploadFailedNotEnoughDiskQuota;
+                break;
+        }
+        userInfo[NSLocalizedDescriptionKey] = errorMessage;
+        newError = [[NSError alloc] initWithDomain:WordPressComApiErrorDomain code:errorCode userInfo:userInfo];
+    } else if (error.domain == WPXMLRPCFaultErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case 500:{
+                errorMessage = NSLocalizedString(@"Your site does not support this media file format.", @"Message to show to user when media upload failed because server doesn't support media type");
+                errorCode = WordPressComApiErrorUploadFailedInvalidFileType;
+            } break;
+            case 401:{
```

Comment:
401 error can happen when the user doesn't have the rights to upload media (Subscriber role for instance).